Complete-TS-12

numerouno-life opened the pull request.

numerouno-life committed:
Complete-TS-12

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -1,84 +1,79 @@
+    public void addLikeToFilm(Film filmId, User userId) {
+        log.info("Add like for film {} by the user {}", filmId, userId);
+        userStorage.getUserById(userId.getId());
+        filmStorage.getFilmById(filmId.getId());
```
этот метод не бросит исключение, если нет такого фильма.

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -1,84 +1,79 @@
-            throw new RuntimeException(msg);
-        }
-        film.getLikes().remove(userId);
+    public boolean removeLike(Long filmId, Long userId) {
```
в этом методе тоже нужно добавить проверки существования фильма и пользователя

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/storage/mpa/MpaDbStorage.java:
```diff
@@ -0,0 +1,33 @@
+
+import java.util.Collection;
+
+@Slf4j
```
в этом классе нет логирования

numerouno-life committed:
Complete-TS-12
Убрал лишнюю аннотацию в MpaDbStorage
добавил проверку валидности в FilmService, в методах addLikeToFilm и removeLike
